packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -37,6 +63,13 @@ typedef ContextMenuPreviewBuilder = Widget Function(
   Widget child,
 );
 
+/// A function that builds the child and handles the transition between the
+/// default child and the preview when the CupertinoContextMenu is open.
+typedef CupertinoContextMenuBuilder = Widget Function(
```

Comment:
Nit: Can you double check that there is no existing typedef that would work for this?

Reply:
There's one that's similar called `AnimatedListRemovedItemBuilder` in `animated_scroll_view`, as well as the one directly under this one, `_ContextMenuPreviewBuilderChildless`. Combining the two ContextMenu builders makes sense to me, but maybe in another PR making a general builder typedef for this and animated scroll view would make sense?

Reply:
Ah I see.  Ok for now I would do this: 1. keep CupertinoContextMenuBuilder 2. get rid of _ContextMenuPreviewBuilderChildless and replace it with CupertinoContextMenuBuilder and 3. deprecate ContextMenuPreviewBuilder.

On the topic of combining CupertinoContextMenuBuilder and AnimatedListRemovedItemBuilder, definitely don't mess with it in this PR as you said, but in the long run I'm not sure what Flutter's policy on this kind of thing is.  It might be worth asking on Discord and/or opening an issue about it.  I wouldn't be surprised if we have other duplicate typedefs with different names like this.  It does make sense to have something like an AnimationBuilder type for both though.